pdns/lua-base4.cc

```diff
@@ -172,8 +172,8 @@ void BaseLua4::prepareContext() {
 
   // NetmaskGroup
   d_lw->writeFunction("newNMG", []() { return NetmaskGroup(); });
-  d_lw->registerFunction<void(NetmaskGroup::*)(const std::string&mask)>("addMask", [](NetmaskGroup&nmg, const std::string& mask) { nmg.addMask(mask); });
-  d_lw->registerFunction<void(NetmaskGroup::*)(const vector<pair<unsigned int, std::string>>&)>("addMasks", [](NetmaskGroup&nmg, const vector<pair<unsigned int, std::string>>& masks) { for(const auto& mask: masks) { nmg.addMask(mask.second); } });
+  d_lw->registerFunction<NetmaskGroup(NetmaskGroup::*)(const std::string&mask)>("addMask", [](NetmaskGroup&nmg, const std::string& mask) { nmg.addMask(mask); return nmg; });
```

Comment:
I think this also affects dnsdist and the recursor? So those docs need to be updated as well.

Reply:
It does not affect dnsdist (it would be nice if it would, but that's clearly out of scope for this PR ;)).

Reply:
Do we intend on returning the altered NMG or a copy of the altered NMG?

Reply:
The intention is to keep the one object that newNMG made. I see I lost a `&` on the left hand side somewhere in the process, I'll check what is happening.

Reply:
Turns out we add the mask, and then we copy the NMG. I'll fix it.

Reply:
I couldn't make LuaWrapper pass the reference around, so I've undone my changes, and added an optional list of masks to `newNMG` as a parameter. I also synced the recursor docs.

Reply:
I remember having to switch to a raw pointer instead of a reference for that reason, a long time ago, for some of dnsdist bindings.
I like the optional list of masks even better, so that's great :-)